CodeEdit/Documents/WorkspaceDocument.swift

```diff
@@ -162,7 +163,7 @@ final class WorkspaceDocument: NSDocument, ObservableObject, NSToolbarDelegate {
     ]
 
     override class var autosavesInPlace: Bool {
-        return false
+        return AppPreferencesModel.shared.preferences.general.isAutoSaveOn
```

Comment:
You can actually remove the `return` keyword as it is redundant in this context.

```swift
override class var autosavesInPlace: Bool {
    AppPreferencesModel.shared.preferences.general.isAutoSaveOn
}
```